clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/consumer/HonoKafkaConsumer.java

```diff
@@ -855,24 +868,28 @@ private Future<Void> subscribeAndWaitForRebalance() {
             topics.forEach(topic -> HonoKafkaConsumerHelper.partitionsFor(kafkaConsumer, topic)
                     .onSuccess(partitions -> {
                         if (partitions.isEmpty()) {
-                            LOG.info("subscription topic doesn't exist and didn't get auto-created: {} [client-id: {}]",
-                                    topic, getClientId());
+                            // either auto-creation of topics is disabled or the (manually) created topic isn't reflected in the result yet
+                            LOG.info("subscription topic doesn't exist as of now: {} [client-id: {}]", topic, getClientId());
                         }
                     }));
             kafkaConsumer.subscribe(topics, subscriptionUpdated);
         }
         // init kafkaConsumerWorker if needed; it has to be retrieved after the first "subscribe" invocation
         if (kafkaConsumerWorker == null) {
             kafkaConsumerWorker = getKafkaConsumerWorker(kafkaConsumer);
+        } else {
+            // not the first "subscribe" invocation;
```

Comment:
when is this supposed to happen? I do not quite understand if this is a branch that we will regularly run into or if this would be the result of *mis-using* the API ...

Reply:
The `subscribeAndWaitForRebalance` method is invoked in two kinds of scenarios:
- once on initialization of the consumer; lifecycleStatus is `starting`,
- in the `ensureTopicIsAmongSubscribedTopicPatternTopics` method (invoked as part of the Command Router `register-cmd-consumer` handling); lifecycleStatus is `started` here.

Therefore, the `// not the first "subscribe" invocation;` branch above will only be called in case of the `ensureTopicIsAmongSubscribedTopicPatternTopics` invocation.

I'm currently working on a further refactoring, also making `ensureTopicIsAmongSubscribedTopicPatternTopics` simpler. This shall make the timer handling above unnecessary.